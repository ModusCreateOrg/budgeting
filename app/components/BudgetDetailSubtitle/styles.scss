@import 'theme/variables';

%description {
  margin-top: 10px;
}

.inflowDescription {
  @extend %description;
  color:$green;
}

.outflowDescription {
  @extend %description;
  color:$red;
}
