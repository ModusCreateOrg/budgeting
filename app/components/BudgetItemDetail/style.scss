@import 'theme/variables';

.budgetItemDetail {
  margin: 5px;

  h1 {
    font-size: 2em;
    margin-bottom: 0px;
  }
  h2 {
    font-size: 1.7em;
    margin-top: 0px;
  }


  @media only screen and (max-width: $screen-small) {
    h1, h2 {
      text-align: center;
    }
  }

}

.income {
  color:$green;
  font-size: 1.1em;
}

.outcome {
  color:$red;
  font-size: 1.1em;
}
