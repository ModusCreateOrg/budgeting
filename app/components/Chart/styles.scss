.mainSvg {
  filter: drop-shadow( 0 1px 3px rgba(0,0,0,.16) );
}
