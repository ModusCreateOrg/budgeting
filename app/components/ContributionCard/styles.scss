$inflowColor1: #58b141;
$inflowColor2: #398026;
$outflowColor1: #EA3324;
$outflowColor2: #981E13;

.heading {
  text-align: center;
  margin-bottom: 0;
}

.percentage {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 5px 0 20px 0;
  font-size: 1.2em;
}

.inflow, .outflow {
  margin-right: 2px;
  font-size: 1.3em;
}

.inflow {
  color: #58b141;
}

.outflow {
  color: #D84433;
}

:export {
  inflowColor1: $inflowColor1;
  inflowColor2: $inflowColor2;
  outflowColor1: $outflowColor1;
  outflowColor2: $outflowColor2;
}
