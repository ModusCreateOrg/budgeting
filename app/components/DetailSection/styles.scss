.detail {
  display: flex;
  flex-flow: column;
  flex: 1;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
}

.heading {
  text-align: center;
}

.inflow {
  color: green;
}

.outflow {
  color: red;
}