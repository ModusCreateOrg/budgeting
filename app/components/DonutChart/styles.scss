.donutChart {
  display: flex;
  justify-content: center;
  align-content: flex-start;
  flex-flow: wrap;  
}
