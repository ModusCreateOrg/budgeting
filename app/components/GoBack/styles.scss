.containerGoBack {
  cursor: pointer;
  display: inline;
  border: 1px solid black;
  width: 70px;
  margin: 5px;
}