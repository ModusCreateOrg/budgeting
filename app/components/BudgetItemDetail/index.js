// @flow
import * as React from 'react';
import type { Transaction } from 'modules/transactions';
import PieChart from 'components/PieChart';

import styles from './style.scss';

type BudgetItemDetailProps = {
  transaction: Transaction,
  balance: number,
};

const BudgetItemDetail = ({ transaction, balance }: BudgetItemDetailProps) => {
  const { value, description } = transaction;

  // Calculate the values considering positive and negative values.
  const contribPercentage = value / balance * 100;
  const remainingBalance = Math.abs(balance) - Math.abs(value);
  const remainingPercentage = 100 - contribPercentage;

  // This flag will be handy to switch styles and signs.
  const isIncome = value >= 0;

  // Create data that Pie Chart needs to render, this could be moved to a helper.
  const data = [
    {
      categoryId: 'item',
      category: `${contribPercentage.toFixed(2)}% - ${description}`,
      value: Math.abs(value),
    },
    {
      categoryId: 'total',
      category: `${remainingPercentage.toFixed(2)}% - Other ${isIncome ? 'incomes' : 'outcomes'}`,
      value: Math.abs(remainingBalance),
    },
  ];

  return (
    <div className={styles.budgetItemDetail}>
      <h1>{transaction.description}</h1>
      <h2>
        <span className={isIncome ? styles.income : styles.outcome}>{isIncome ? '+' : '-'}</span>
        {contribPercentage.toFixed(2)}%
      </h2>
      <PieChart data={data} dataLabel="category" dataKey="categoryId" />
    </div>
  );
};

export default BudgetItemDetail;
