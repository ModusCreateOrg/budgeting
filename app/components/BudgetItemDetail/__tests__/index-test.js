import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import BudgetItemDetail from 'components/BudgetItemDetail';
import styles from '../style.scss';

jest.mock('components/PieChart/Path');
jest.mock('components/PieChart', () => 'MockPieChart');

function setup(isOutflow = false) {
  const transaction = {
    id: 1,
    categoryId: 1,
    description: 'Mock testing description',
    value: isOutflow ? -33.33 : 33.33,
  };

  const balance = isOutflow ? -100 : 100;

  const props = {
    transaction,
    balance,
  };
  const enzymeWrapper = shallow(<BudgetItemDetail {...props} />);

  return {
    enzymeWrapper,
    props,
  };
}

it('renders correctly', () => {
  const { props } = setup();
  const tree = renderer.create(<BudgetItemDetail {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render title and subtitle', () => {
  const { enzymeWrapper } = setup();
  expect(enzymeWrapper.find('h1').text()).toBe('Mock testing description');
  expect(enzymeWrapper.find('h2').exists()).toBe(true);
});

it('should render subtitle percentage, sign and color correctly', () => {
  const inflowWrapper = setup().enzymeWrapper;
  expect(inflowWrapper.find('h2').text()).toBe('+33.33%');
  expect(
    inflowWrapper
      .find('h2')
      .find('span')
      .hasClass(styles.income)
  ).toBe(true);

  const outflowWrapper = setup(true).enzymeWrapper;
  expect(outflowWrapper.find('h2').text()).toBe('-33.33%');
  expect(
    outflowWrapper
      .find('h2')
      .find('span')
      .hasClass(styles.outcome)
  ).toBe(true);
});

it('should render a Pie Chart component', () => {
  const { enzymeWrapper } = setup();
  expect(enzymeWrapper.find('MockPieChart').exists()).toBe(true);
});
