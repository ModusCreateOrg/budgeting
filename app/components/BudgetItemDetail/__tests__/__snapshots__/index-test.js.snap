// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="budgetItemDetail"
>
  <h1>
    Mock testing description
  </h1>
  <h2>
    <span
      className="income"
    >
      +
    </span>
    33.33
    %
  </h2>
  <MockPieChart
    data={
      Array [
        Object {
          "category": "33.33% - Mock testing description",
          "categoryId": "item",
          "value": 33.33,
        },
        Object {
          "category": "66.67% - Other incomes",
          "categoryId": "total",
          "value": 66.67,
        },
      ]
    }
    dataKey="categoryId"
    dataLabel="category"
  />
</div>
`;
